stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModelTest.kt

```diff
@@ -2,48 +2,65 @@
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Command
+import io.getstream.chat.android.client.models.Config
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.offline.plugin.state.StateRegistry
+import io.getstream.chat.android.offline.plugin.state.channel.ChannelState
+import io.getstream.chat.android.test.InstantTaskExecutorExtension
+import io.getstream.chat.android.test.asCall
+import io.getstream.chat.android.test.randomCID
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
 @ExtendWith(InstantTaskExecutorExtension::class)
 internal class MessageInputViewModelTest {
 
     private val CID = randomCID()
-    private val chatDomain: ChatDomain = mock()
-    private val chatClient: ChatClient = mock()
-
-    @Suppress("DEPRECATION_ERROR")
-    private val channelControllerResult: Result<ChannelController> = mock()
-    private val channelControllerCall = TestCall(channelControllerResult)
-    private val channelController: ChannelController = mock()
-    private val commands: List<Command> = createCommands()
-    private val channel: Channel = createChannel(cid = CID, config = Config(commands = commands))
-
-    @BeforeEach
-    fun setup() {
-        whenever(chatDomain.watchChannel(eq(CID), eq(0))) doReturn channelControllerCall
-        whenever(chatDomain.user) doReturn MutableLiveData(randomUser())
-        whenever(channelController.toChannel()) doReturn channel
-        whenever(channelController.offlineChannelData) doReturn MutableLiveData(mock())
-        whenever(channelControllerResult.isSuccess) doReturn true
-        whenever(channelControllerResult.data()) doReturn channelController
+    private val channelConfig: Config = mock {
+        whenever(it.commands) doReturn createCommands()
     }
 
-    @Test
-    fun `Should show members`() {
-        val members = createMembers()
-        whenever(channelController.members) doReturn MutableLiveData(members)
-        val messageInputViewModel = MessageInputViewModel(CID, chatDomain, chatClient)
-        val mockObserver: Observer<List<Member>> = spy()
-        messageInputViewModel.members.observeForever(mockObserver)
-
-        verify(mockObserver).onChanged(eq(members))
+    private val channelState: ChannelState = spy {
+        whenever(it.repliedMessage) doReturn MutableStateFlow(Message())
+        whenever(it.channelConfig) doReturn MutableStateFlow(channelConfig)
+        whenever(it.members) doReturn MutableStateFlow(createMembers())
+    }
+
+    private val stateRegistry: StateRegistry = mock {
+        whenever(it.channel(any(), any())) doReturn channelState
```

Comment:
does it make sense to call mock and then in the provider use a `whenever`? Shouldn't a `mock` just return a mock instance and then in the setup, e.g. the test itself, we should set up any required `whenever` blocks?

If it's required in _all_ tests, then that's why we have an `init` and `@Before` block.